common/http/platform/beast/http.cpp

```diff
@@ -706,6 +710,20 @@ void Client::ReadBodyHandler(error_code ec, size_t num_read) {
 }
 
 void Client::Cancel() {
+	if (client_active_) {
+		auto err =
+			error::Error(make_error_condition(errc::operation_canceled), "HTTP request cancelled");
+		if (status_ == TransactionStatus::None) {
+			CallErrorHandler(err, request_, header_handler_);
+		} else if (status_ != TransactionStatus::Done) {
+			CallErrorHandler(err, request_, body_handler_);
+		}
```

Comment:
Are we sure that `Cancel` can't be called in other status? What about between body parts read? Then the status would be `TransactionStatus::BodyReadingInProgress` and it is not caught here. Is that handled from somewhere else?

Reply:
Hmm, maybe you misread it, it's all statuses *except* `Done`, so they should all be caught here.

Reply:
I misread :see_no_evil: